File: src/main/java/gg/mineral/practice/commands/kit/LeaveCommand.java
package gg.mineral.practice.commands.kit;

import gg.mineral.practice.commands.PlayerCommand;
import gg.mineral.practice.entity.PlayerStatus;
import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.util.messages.impl.ErrorMessages;
import lombok.val;

public class LeaveCommand extends PlayerCommand {

    public LeaveCommand() {
        super("leave");
    }

    @Override
    public void execute(org.bukkit.entity.Player player, String[] args) {
        val profile = ProfileManager.getOrCreateProfile(player);

        if (profile.getPlayerStatus() != PlayerStatus.KIT_CREATOR
                || profile.getPlayerStatus() != PlayerStatus.KIT_EDITOR) {
            profile.message(ErrorMessages.NOT_IN_KIT_EDITOR_OR_CREATOR);
            return;
        }

        if (profile.getPlayerStatus() == PlayerStatus.KIT_CREATOR) {
            profile.leaveKitCreator();
            return;
        }

        profile.leaveKitEditor();
    }

}

File: src/main/kotlin/gg/mineral/practice/commands/kit/LeaveCommand.kt
package gg.mineral.practice.commands.kit

import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.PlayerStatus
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.util.messages.impl.ErrorMessages

@Command(name = "leave")
class LeaveCommand {
    @Execute
    fun execute(@Context profile: Profile) {
        if (profile.playerStatus !== PlayerStatus.KIT_CREATOR
            && profile.playerStatus !== PlayerStatus.KIT_EDITOR
        ) {
            profile.message(ErrorMessages.NOT_IN_KIT_EDITOR_OR_CREATOR)
            return
        }

        profile.kitCreator = null
        profile.kitEditor = null
    }
}
